packages/preview-middleware/src/base/flp.ts

```diff
@@ -174,6 +174,40 @@ export class FlpSandbox {
         this.logger.debug(`Configured apps: ${JSON.stringify(this.templateConfig.apps)}`);
     }
 
+    /**
+     * Gets the FLP sandbox for an editor.
+     *
+     * @param _req (not used) original request
+     * @param res response of the request
+     * @param rta runtime authoring configuration
+     * @param editor editor configuration
+     */
+    private async getSandboxForEditor(_req: Request, res: Response, rta: RtaConfig, editor: Editor) {
```

Comment:
I assume the extraction of the code into a separate function was done to reduce the cognitive complexity, however, if we extract it, let's do it slightly differently to keep the functionality better encapsulated.
Suggestion: have a`private async function generateSandboxForEditor(rta: RtaConfig, editor: Editor): Promise<string>` that is returning the templated `html` and then keep in the calling function

```
this.router.get(previewUrl, (async (req: Request, res: Response) => {
  const html = await this.generateSandboxForEditor(rta, editor);
  res.status(200).contentType('html').send(html);
```

As a result you have nicely separated the templating and the request handling.

Reply:
Thanks Tobias, will do.

Reply:
Done with https://github.com/SAP/open-ux-tools/pull/1363/commits/5fde330ca07652a4719a9239001c3e336b7f8f34

Reply:
Perfect. Btw, I just noticed that there is no `await` anymore, so we could actually remove the async from the handler and remove the type case, right?

Reply:
Right, done with https://github.com/SAP/open-ux-tools/pull/1363/commits/f8d3e76fae834a14ab2c303dabd4845f5b558b77